src/Package.php

```diff
@@ -53,7 +53,9 @@ public static function init() {
 			self::$active_version = WC_ADMIN_VERSION_NUMBER;
 			$update_version       = new WC_Admin_Notes_Deactivate_Plugin();
 			if ( version_compare( WC_ADMIN_VERSION_NUMBER, self::VERSION, '<' ) ) {
-				$update_version::possibly_add_note();
+				if ( method_exists( $update_version, 'possibly_add_note' ) ) {
```

Comment:
This is a solid fix and works well to fix the issue.

This is mostly my fault, but one of the main issues here is that we're mixing two different versions of WCA every time we do this.  I think going forward we have 2 approaches to solve this:

1. Pull the note logic into `Package.php`.  This is messy, but guarantees we always have the note.
1. Add an action here (e.g., `do_action( 'woocommerce_admin_package_override', WC_ADMIN_VERSION_NUMBER,  self::VERSION )`.

This way, no shared knowledge of classes exists between the 2 different versions.

Reply:
Great suggestion Josh. I think given the timeline for this release cycle we should ship this fix as-is, and refactor during the next cycle.